api/desecapi/migrations/0041_rename_user_token_owner_token_user_override_and_more.py

```diff
@@ -0,0 +1,98 @@
+# Generated by Django 5.1.3 on 2024-11-25 19:53
+
+import django.db.models.deletion
+import django.db.models.functions.comparison
+from django.conf import settings
+from django.db import migrations, models
+
+
+class Migration(migrations.Migration):
+
+    dependencies = [
+        ("desecapi", "0040_token_auto_policy_token_token_auto_policy_and_more"),
+    ]
+
+    operations = [
+        migrations.RunSQL(
+            sql="ALTER TABLE desecapi_tokendomainpolicy DROP CONSTRAINT desecapi_tokendomainpolicy_token_id_token_user_id_fkey RESTRICT;",
+            reverse_sql="",
+        ),
+        migrations.RemoveConstraint(
+            model_name="token",
+            name="unique_id_user",
+        ),
+        migrations.AlterField(
+            model_name="token",
+            name="user",
+            field=models.UUIDField(),
+        ),
+        migrations.RenameField(
+            model_name="token",
+            old_name="user",
+            new_name="owner",
+        ),
+        migrations.AlterField(
+            model_name="token",
+            name="owner",
+            field=models.ForeignKey(
+                on_delete=django.db.models.deletion.CASCADE,
+                related_name="+",
+                to=settings.AUTH_USER_MODEL,
+            ),
+        ),
+        migrations.AddField(
+            model_name="token",
+            name="user_override",
+            field=models.ForeignKey(
+                null=True,
+                on_delete=django.db.models.deletion.CASCADE,
```

Comment:
add documentation

Reply:
Will do. For the record:

The "naive" implementation gives the behavior that
- deletion of the impersonated user also deletes the impersonation token (insead of returning it do its owner). This is due to `on_delete=CASCADE`;
- to detach an impersonation token from the target account, the target (effective) user can simply delete it. This is "the normal behavior" for token management.

Impersonation tokens can only be managed by the target (effective) user, not by their owner. Both users however can see the token and its configuration.

We will expose `user_override` in the token serializer, and will allow writing an account identifier (likely the email address) into it in order to request approval from the target user. We need to ensure that the only allowed transition is "from NULL to user" (but not "from user to NULL" or "from user1 to user2").

Shall document all of this.

Reply:
done.